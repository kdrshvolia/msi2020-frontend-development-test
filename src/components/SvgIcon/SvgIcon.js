import React from 'react';
import PropTypes from 'prop-types';

const SvgIcon = ({ fill, width, height, viewBox = '0 0 20 17', path, onClick }) => (
  <svg onClick={onClick} width={width} height={height} viewBox={viewBox}>
    <path d={path} fill={fill} />
  </svg>
);

SvgIcon.propTypes = {
  fill: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
  height: PropTypes.string.isRequired,
  viewBox: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default SvgIcon;
