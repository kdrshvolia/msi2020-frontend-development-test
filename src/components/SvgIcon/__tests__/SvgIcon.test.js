import React from 'react';import renderer from 'react-test-renderer';import SvgIcon from '../SvgIcon';describe('SvgIcon', () => {  it('renders correctly', () => {    const isClicked = jest.fn();    const rendered = renderer.create(      <SvgIcon        viewBox="0 0 20 17"        onClick={isClicked}        height="20"        width="20"        path="2"        fill="red"      />,    );    expect(rendered.toJSON()).toMatchSnapshot();  });});