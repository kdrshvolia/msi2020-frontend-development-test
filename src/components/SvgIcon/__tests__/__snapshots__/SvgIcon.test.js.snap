// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon renders correctly 1`] = `
<svg
  height="20"
  onClick={[MockFunction]}
  viewBox="0 0 20 17"
  width="20"
>
  <path
    d="2"
    fill="red"
  />
</svg>
`;
